havoc.c

```diff
@@ -135,16 +135,16 @@ int test_havoc(){
           return ++errors;
         }
 
+        if (verbose)
+            Printf("Process %d changing allocation : %d -> %d, current allocation: %d%% \n", pid, processes[pid].request, new_allocation, (100-remaining_allocation - processes[pid].request + new_allocation));
```

Comment:
After I tried to debug my code last night, I found it useful to know what is the current allocation. It might help to recognize some edge cases.